src/main/java/view/StudySessionView.java

```diff
@@ -214,6 +232,12 @@ else if (evt.getSource().equals(stopTimerButton)) {
             studySessionController.stopStudyTimer();
 
         }
+        else if (evt.getSource().equals(playPauseButton)) {
+            // Toggle playback
+            musicControlController.togglePlayback();
+            // Update button text based on the new state
+            playPauseButton.setText(musicControlViewModel.getState().isPlaying() ? "Pause" : "Play");
```

Comment:
Again, replace with proper if and else loop. the ? "Pause" : "Play" is not recommended.